From 0e125202bf0d7bb75af28016ad7648fd3c1c2e4f Mon Sep 17 00:00:00 2001
From: Jun Li <r65092@freescale.com>
Date: Fri, 24 Oct 2008 17:51:04 +0800
Subject: [PATCH] ENGR00096583 MX51 USB OTG function support

Add USB OTG function for USB dr port in i.MX51

Signed-off-by: Jun Li <r65092@freescale.com>
---
 arch/arm/mach-mx51/usb_dr.c |   19 ++++++++++++++++++-
 1 files changed, 18 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-mx51/usb_dr.c b/arch/arm/mach-mx51/usb_dr.c
index be303b7..036bd51 100644
--- a/arch/arm/mach-mx51/usb_dr.c
+++ b/arch/arm/mach-mx51/usb_dr.c
@@ -57,6 +57,11 @@ static void dr_udc_release(struct device *dev)
 {
 }
 
+static u64 dr_otg_dmamask = ~(u32) 0;
+static void dr_otg_release(struct device *dev)
+{
+}
+
 /*
  * platform device structs
  * 	dev.platform_data field plugged at run time
@@ -73,6 +78,18 @@ static struct platform_device dr_udc_device = {
 	.num_resources = ARRAY_SIZE(otg_resources),
 };
 
+static struct platform_device __maybe_unused dr_otg_device = {
+	.name = "fsl-usb2-otg",
+	.id = -1,
+	.dev = {
+		.release           = dr_otg_release,
+		.dma_mask          = &dr_otg_dmamask,
+		.coherent_dma_mask = 0xffffffff,
+	},
+	.resource      = otg_resources,
+	.num_resources = ARRAY_SIZE(otg_resources),
+};
+
 /* Notes: configure USB clock*/
 static int usbotg_init_ext(struct platform_device *pdev)
 {
@@ -113,7 +130,7 @@ static int __init usb_dr_init(void)
 {
 	pr_debug("%s: \n", __func__);
 
-	/* dr_register_otg(); */
+	dr_register_otg();
 	dr_register_host(otg_resources, ARRAY_SIZE(otg_resources));
 	dr_register_udc();
 
-- 
1.5.4.4

